File: 344 Reverse String/reverseString.py
class Solution(object):
    def reverseString(self, s):
        """
        :type s: str
        :rtype: str
        """
        if len(s) == 0 or len(s) == 1:
            return s
        else:
            l = 0
            r = len(s) - 1

            # Note that in Python, string is unchangable so we need to convert it to a list
            stemp = list(s)
            while l < r:
                stemp[l],stemp[r] = stemp[r],stemp[l]
                l += 1
                r -= 1
            return "".join(stemp)

File: 344 Reverse String/reverseString.cpp
class Solution {
public:
    string reverseString(string s) {
        if(s.size() == 0)
            return s;
        else{
            int l = 0;
            int r = s.size() - 1;
            
            while(l < r){
                swap(s[l++],s[r--]);
            }
            if(l >= r)
                return s;
        }
    }
};